privacyidea/lib/eventhandler/requestmangler.py

```diff
@@ -0,0 +1,163 @@
+# -*- coding: utf-8 -*-
+#
+#  2019-09-06 Cornelius Kölbel <cornelius.koelbel@netknights.it>
+#             Initial writup
+#
+# License:  AGPLv3
+# (c) 2019. Cornelius Kölbel
+#
+# This code is free software; you can redistribute it and/or
+# modify it under the terms of the GNU AFFERO GENERAL PUBLIC LICENSE
+# License as published by the Free Software Foundation; either
+# version 3 of the License, or any later version.
+#
+# This code is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU AFFERO GENERAL PUBLIC LICENSE for more details.
+#
+# You should have received a copy of the GNU Affero General Public
+# License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+__doc__ = """This is the event handler module modifying request parameters.
+"""
+from privacyidea.lib.eventhandler.base import BaseEventHandler
+from privacyidea.lib.token import (get_token_types, set_validity_period_end,
+                                   set_validity_period_start)
+from privacyidea.lib.realm import get_realms
+from privacyidea.lib.token import (set_realms, remove_token, enable_token,
+                                   unassign_token, init_token, set_description,
+                                   set_count_window, add_tokeninfo,
+                                   set_failcounter, delete_tokeninfo)
+from privacyidea.lib.utils import (parse_date, is_true,
+                                   parse_time_offset_from_now)
+from privacyidea.lib.tokenclass import DATE_FORMAT, AUTH_DATE_FORMAT
+from privacyidea.lib import _
+import json
+import logging
+import re
+import datetime
+import yaml
+from dateutil.parser import parse as parse_date_string
+from dateutil.tz import tzlocal
+
+log = logging.getLogger(__name__)
+
+
+class ACTION_TYPE(object):
+    """
+    Allowed actions
+    """
+    ADD = "add"
```

Comment:
As `ADD` also overwrites existing entries, what do you think about renaming it to ``SET``?

Reply:
OK, I understand. Thought about it, too, that it is not always an "add".

Will do

Reply:
done in 4fd308ea